zone.dos.Browser.yaml

```diff
@@ -0,0 +1,67 @@
+id: zone.dos.Browser
+runtime: org.freedesktop.Platform
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '24.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node20
+command: run.sh
+separate-locales: false
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node20/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: dos-browser
+    only-arches:
+      - x86_64
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/dos-browser/flatpak-node/cache
+        npm_config_cache: /run/build/dos-browser/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+    build-commands:
+      # I have no idea why node-gyp from generated-sources.json is not working.
+      # https://github.com/flatpak/flatpak-builder/issues/629
+      - unzip node-gyp.zip -d ./
+      - npm install --offline
+      - npm run make -- --targets "@electron-forge/maker-zip"
+      - unzip out/make/zip/linux/x64/DOS.Zone\ Browser-linux-x64-8.0.5.zip -d /app
+      - mv /app/DOS.Zone\ Browser-linux-x64 /app/browser
+      - mv doszone-backend /app/doszone-backend && chmod 777 /app/doszone-backend
+      - mv doszone-backend-x /app/doszone-backend-x && chmod 777 /app/doszone-backend-x
+      - install -Dm755 -t /app/bin/ run.sh
+      - install -Dm644 flathub/zone.dos.Browser.desktop /app/share/applications/zone.dos.Browser.desktop
+      - install -Dm644 flathub/zone.dos.Browser.metainfo.xml /app/share/metainfo/zone.dos.Browser.metainfo.xml
+      - install -Dm644 flathub/zone.dos.Browser.png /app/share/icons/hicolor/512x512/apps/zone.dos.Browser.png
+    sources:
+      # - type: dir
+      #   path: ..
+      - type: git
+        url: https://github.com/js-dos/dos.zone.browser.pc
+        commit: dada4c62326a2822afdf1199387e686cfe14f637
+      - type: file
+        url: https://v8.js-dos.com/flatpak/node-gyp.zip
+        sha256: 8905e15b377f3e26e1515d48f1b316b7deca004654c2e52239d599799cb4976f
+      - type: file
+        url: https://v8.js-dos.com/flatpak/v8.0.5/doszone-backend
+        sha256: cd2089777e22f32ee645e79528ff74cab7c069af7920ef1a1fee4eb98badb219
+      - type: file
+        url: https://v8.js-dos.com/flatpak/v8.0.5/doszone-backend-x
+        sha256: c1108f6f33a7cddec16102c03b7aa2052fcf543ee7266ded594abf0425d3f2c2
+      # The linter gives a warning here (module-dos-browser-source-git-url-not-http). 
+      # If I use file instead, the build stops working.
+      # Also in tutorial it's used like this.
+      - generated-sources.json
```

Comment:
you have change the git+ssh url here to `http(s)`.

Reply:
this one

```
    {
        "type": "git",
        "url": "git+ssh://git@github.com/electron/node-gyp.git",
        "commit": "06b29aafb7708acef8b3669835c8a7857ebc92d2",
        "dest": "flatpak-node/git-packages/@electron/node-gyp-06b29aafb7708acef8b3669835c8a7857ebc92d2"
    },
```

Reply:
your package-lock.json should use http too. not sure why it is using ssh URLs.

Reply:
Oh, I regenerated the package-lock.json, and now it contains only HTTPS URLs. I’m not sure why it used git+ssh before; I just found out that it is also valid, but I don’t know the exact rule for when it uses git+ssh versus git+https. Anyway, I also regenerated the generated-sources.json, and now the linter passes.